### PR TITLE
Handle broken relations properly.

### PR DESCRIPTION
**Pull request by mbaechtold**

### CODE REVIEW

File: ftw/publisher/core/utils.py [hunk 1]
```diff
@@ -150,6 +151,19 @@ def decode_for_json(value, additional_encodings=[]):
                            'outputMimeType': value.outputMimeType,
                            'encoding': value.encoding}}
 
+    # RelationValue
+    if isinstance(value, RelationValue):
+        if not value.to_object:
+            return None
```
Comment (jone) [inline ftw/publisher/core/utils.py]: I would add a comment here:
```
# The relation is broken: the target does not exist. This is not fixable.
```

File: ftw/publisher/core/utils.py [hunk 2]
```diff
@@ -275,10 +289,26 @@ def encode_after_json(value):
...
 
+    # RelationValue
+    elif isinstance(value, dict) and value.get('utf8:publisher-wrapper', False) \
+            and value.get('utf8:type', None) == 'utf8:RelationValue':
+
+        data = encode_after_json(value['utf8:data'])
+
+        obj = get_obj_by_relative_path(data['to_path'])
+        if not obj:
+            return None
```
Comment (jone) [inline ftw/publisher/core/utils.py]: A comment here too:
```
# The target object does not exist on the receiver side. We cannot make a relation at this point.
```

File: ftw/publisher/core/utils.py (hunk 1 above)
Comment (mbaechtold) [inline ftw/publisher/core/utils.py]: Thanks. I have added the comment.

File: ftw/publisher/core/utils.py (hunk 2 above)
Comment (mbaechtold) [inline ftw/publisher/core/utils.py]: Thanks. I have added the comment.

**Comment by mbaechtold** [comment]
@jone, I finally found the time to rebase on the `master` branch. While doing so I revised the proposed change and I don't remember what it had to do with nested relations (as proposed in the title of the pull request). Instead, it rather looks like a fix for broken relations.

What do you think?

**Pull request merged**